example_updates/nominal_update/example-a-b-update.sh

```diff
@@ -0,0 +1,768 @@
+    echo "                          See InstallUpdate() function for more details."
+    echo ""
+    echo "--log-level <0-4>         A minimum log level. 0=debug, 1=info, 2=warning, 3=error, 4=none."
+    echo "-h, --help                Show this help message."
+    echo ""
+    echo "Example:"
+    echo ""
+    echo "Scenario: is-installed check"
+    echo "========================================"
+    echo "    <script> --log-level 0 --action-is-installed --intalled-criteria 1.0"
+    echo ""
+    echo "Scenario: perform install action"
+    echo "================================"
+    echo "    <script> --log-level 0 --action-install --intalled-criteria 1.0 --swu-file example-device-update.swu --work-folder <sandbox-folder>"
+    echo ""
+}
+
+log "Log begin:"
+output "Output begin:"
+
+#
+# Parsing arguments
+#
+while [[ $1 != "" ]]; do
+    case $1 in
+
+    #
+    # Device Update specific arguments.
+    #
+    --action-download)
+        shift
+        do_download_action=yes
+        ;;
+
+    --action-install)
+        shift
+        log_info "Will runscript as 'installer' script."
+        do_install_action=yes
+        ;;
+
+    --action-apply)
+        shift
+        do_apply_action=yes
+        ;;
+
+    --restart-to-apply)
+        shift
+        restart_to_apply=yes
+        ;;
+
+    --restart-agent-to-apply)
+        shift
+        restart_agent_to_apply=yes
+        ;;
+
+    --action_cancel)
+        shift
+        do_cancel_action=yes
+        ;;
+
+    --action-is-installed)
+        shift
+        check_is_installed=yes
+        ;;
+
+    --installed-criteria)
+        shift
+        if [[ -z $1 || $1 == -* ]]; then
+            error "--installed-criteria requires an installedCriteria parameter."
+            $ret 1
+        fi
+        installed_criteria="$1"
+        shift
+        ;;
+
+    #
+    # Update artifacts
+    #
+    --swu-file)
+        shift
+        if [[ -z $1 || $1 == -* ]]; then
+            error "--swu-file parameter is mandatory."
+            $ret 1
+        fi
+        image_file="$1"
+        echo "swu file (image_file): $image_file"
+        shift
+        ;;
+
+    --work-folder)
+        shift
+        if [[ -z $1 || $1 == -* ]]; then
+            error "--work-folder parameter is mandatory."
+            $ret 1
+        fi
+        workfolder="$1"
+        echo "work folder: $workfolder"
+        shift
+        ;;
+
+    #
+    # Output-related arguments.
+    #
+    # --out-file <file_path>, --result-file <file_path>, --log-file <file_path>
+    #
+    --output-file)
+        shift
+        if [[ -z $1 || $1 == -* ]]; then
+            error "--out-file parameter is mandatory."
+            $ret 1
+        fi
+        output_file="$1"
+
+        #
+        #Create output file path.
+        #
+        # Delete existing log.
+        rm -f -r "$output_file"
+        # Create dir(s) recursively (include filename, well remove it in the following line...).
+        mkdir -p "$output_file"
```

Comment:
append `|| $ret` in case it fails?  I see lots of lines missing this, which suggests it was not run through shell-check.  Run shell-check on these scripts?